py_zipkin/_encoding_helpers.py

```diff
@@ -64,3 +73,203 @@ def copy_endpoint_with_new_service_name(endpoint, new_service_name):
         ipv6=endpoint.ipv6,
         port=endpoint.port,
     )
+
+
+def get_encoder(encoding):
+    """Creates encoder object for the given encoding.
+
+    :param encoding: desired output encoding protocol.
+    :type encoding: Encoding
+    """
+    if encoding == Encoding.THRIFT:
+        return _V1ThriftEncoder()
+    elif encoding == Encoding.JSON:
+        return _V1JSONEncoder()
+    else:
+        raise ValueError('Unknown encoding')
+
+
+class IEncoder(object):
+    """Encoder interface."""
+
+    def fits(self, queue, current_size, max_size, new_span):
+        """Returns the overhead in bytes of a list.
+
+        The overhead is measured as the difference between
+        the sum of the size in bytes of all objects and the
+        size of their list concatenation.
+
+        i.e. In JSON is 2 since
+
+        :param queue: queue
+        :type queue: list
+        :returns: (int) overhead in bytes of a list object.
+        """
+        raise NotImplementedError()
+
+    def encode_span(self, *argv):
```

Comment:
I think v1 and v2 parameters will be the same. I used *argv here mostly for laziness since I didn't want to repeat that list of 10 arguments again. :)

I thought about using a Span namedtuple or class, but I was trying to minimize the code changes. Let me see how big of a difference that makes.